components/camel-azure/src/main/java/org/apache/camel/component/azure/blob/BlobServiceUtil.java

```diff
@@ -178,7 +179,7 @@ public static CloudBlob getConfiguredClient(Exchange exchange, BlobServiceConfig
             if (client.getClass() != expectedCls) {
                 throw new IllegalArgumentException("Invalid Client Type");
             }
-            if (!client.getUri().equals(prepareStorageBlobUri(exchange, cfg))) {
+            if (validateURI && !client.getUri().equals(prepareStorageBlobUri(exchange, cfg))) {
```

Comment:
A question, is the check of `!client.getUri().equals(prepareStorageBlobUri(exchange, cfg)` even needed? I mean if this is out, you don't need even to have extra configs to disable the URI validation. What others here think? Because the Azure URI segment could change in the future, we never know

Reply:
Yeah I did consider that. But, even if you're using a custom client, it may still be desirable to do some validation, hence I added a new URI param to control it.

Reply:
I see, but for the me the whole point of using a custom client (inherited from the main `CloudBlob`) clients to bypass these kind of validations, isn't?

Reply:
I'm not the author of the component so I can't be sure what the original intent was.

In theory, there are some customizations that you can set on the client without changing the service URL (E.g blob snapshot ID) etc. So in that case the validation logic would still be valid.

I'm not too bothered which solution we opt for. I don't think we do such validation in the new `camel-azure-storage-blob` component do we? So maybe it does make sense to remove it from `camel-azure` for consistency.

Reply:
Nope we don't, when we ported this component to the new v12 APIs, we omitted this part for now as we didn't see the the need to include such checks, IMHO a client if provided, should be agnostic as much as possible from such checks and allows some flexibility. IMHO I am in favour of removing it and simplify the configurations, what others think? e.g @oscerd ?

Reply:
I think, until you don't modify the client, but just doing some validation it is ok. The problem is when you modify the state of the provided client eventually. So to me this is ok.